src/PhpSpreadsheet/Calculation/DateTimeExcel/Datedif.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+namespace PhpOffice\PhpSpreadsheet\Calculation\DateTimeExcel;
+
+use DateInterval;
+use DateTime;
+use Exception;
+use PhpOffice\PhpSpreadsheet\Calculation\Functions;
+use PhpOffice\PhpSpreadsheet\Shared\Date;
+
+class Datedif
+{
+    /**
+     * DATEDIF.
+     *
+     * @param mixed $startDate Excel date serial value, PHP date/time stamp, PHP DateTime object
+     *                                    or a standard date string
+     * @param mixed $endDate Excel date serial value, PHP date/time stamp, PHP DateTime object
+     *                                    or a standard date string
+     * @param string $unit
+     *
+     * @return int|string Interval between the dates
+     */
+    public static function funcDatedif($startDate, $endDate, $unit = 'D')
```

Comment:
Wondering if it's a good idea to create public class constants for the possible `$unit` values; that we can then use in the actual code (lines 49-54); and perhaps a private constant array of the values that can make input validation easier.

The same principle can be applied to fixed range of value arguments for other class/methods like WEEKDAY()

All part of improving the codebase as we go through this exercise

Reply:
I followed your suggestion below of a separate Constants class. This seemed right because WeekDay will want to use the same constants as WeekNum when we get to implementing the new "style" values. All constants in DateTime are now marked deprecated following the model in Engineering.